src/main/java/com/jakub/todoSandbox/repository/JOOQRepository.java

```diff
@@ -0,0 +1,224 @@
+package com.jakub.todoSandbox.repository;
+
+
+import com.jakub.todoSandbox.jooq.tables.records.StepRecord;
+import com.jakub.todoSandbox.model.Priority;
+import com.jakub.todoSandbox.model.Step;
+import com.jakub.todoSandbox.model.Todo;
+import com.jakub.todoSandbox.model.ValidationException;
+import org.jooq.DSLContext;
+import org.jooq.Result;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import static com.jakub.todoSandbox.jooq.tables.Todo.TODO;
+import static com.jakub.todoSandbox.jooq.tables.Step.STEP;
+
+import com.jakub.todoSandbox.jooq.enums.PriorityEnum;
+import com.jakub.todoSandbox.jooq.tables.records.TodoRecord;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Repository
+public class JOOQRepository implements TodoRepository {
+
+    private final DSLContext context;
+
+    public JOOQRepository(DSLContext context) {
+        this.context = context;
+    }
+
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
+    @Override
+    public Optional<Todo> findTodoById(long todoId) {
+        System.out.println("findTodoById() method call with todoId: {" + todoId + "}.");
+        return context.selectFrom(TODO)
+                .where(TODO.ID.eq(todoId))
+                .fetchOptional()
+                .map(record -> Todo.builder(record.getName())
+                        .id(record.getId())
+                        .description(record.getDescription())
+                        .priority(Priority.valueOf(record.getPriority().name()))
+                        .steps(fetchSteps(record.getId()))
+                        .build());
+    }
+
+    @Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
+    @Override
+    public List<Todo> findAllTodos() {
+        System.out.println("findAllTodos() method call");
+        Comparator<Todo> priorityComparator = Comparator.comparing(Todo::priority);
+        return context.selectFrom(TODO)
+                .fetch()
+                .map(record -> Todo.builder(record.getName())
+                        .id(record.getId())
+                        .description(record.getDescription())
+                        .priority(Priority.valueOf(record.getPriority().name()))
+                        .steps(fetchSteps(record.getId()))
+                        .build())
+                .stream()
```

Comment:
You fetch a list of todos and then for each todo you fetchSteps. It's a typical N + 1 query problem, there will be lots of queries, fix it ;)